### ISSUE TITLE
Valid use of calc in media queries results in an error

### ISSUE THREAD

**Issue by dumptyd** [csstree/csstree#162]
```css
@media (max-width: calc(400px + 1px)) {
  .btn { width: 100px; }
}
```
```diff
- Parse error: Number, dimension, ratio or identifier is expected
```

That code can be ran here to reproduce the error: https://csstree.github.io/docs/validator.html

**Comment by lahmatiy** [comment]
The parsing of media queries are not complete (see https://github.com/csstree/csstree/issues/118). This is not an easy question and I am still wondering what to do about it.

However, that's quite a surprise to me that `calc()` is allowed in media query. As I can see, browsers don't support it and I have not seen in any CSS spec that this is allowed. Are you sure that's valid?

**Comment by abdo643-HULK** [comment]
It is allowed and most browsers support it. 
https://stackoverflow.com/questions/23667208/calc-not-working-within-media-queries

**Comment by lahmatiy** [comment]
@abdo643-HULK Wow, didn't know that. Thank you for the link 🙏

**Comment by Mrten** [comment]
Stumbled over this, but the stackoverflow answer has been changed:

> ANSWER EDITED AGAIN 21.03.2022:
>
> In the current version of the spec, using calc (or var) in media queries is NOT supported by [the spec](https://www.w3.org/TR/mediaqueries-4/#mq-features) (as TylerH pointed out below).```

**Comment by epmatsw** [comment]
And interesting problem. https://chromestatus.com/feature/5643732019838976 indicates that Chrome at least wants this to be supported.

**Comment by romainmenke** [comment]
The stackoverflow answer is incorrect.
This is definitely required to work by the relevant specifications.

https://stackoverflow.com/a/74961202/4263818

**Comment by lahmatiy** [comment]
Fixed in [v3.0.0](https://github.com/csstree/csstree/releases/tag/v3.0.0)
